build_tools/compiler/template_processor.rb

```diff
@@ -39,5 +40,71 @@ def asset_path(file, options={})
     def method_missing(name, *args)
       puts "#{name} #{args.inspect}"
     end
+
+    def stylesheet_link_tag(*sources)
+      options = stringify_keys(extract_options!(sources))
+      sources.uniq.map { |source|
+        link_options = {
+            "rel" => "stylesheet",
+            "media" => "screen",
+            "href" => asset_path(source)
+        }.merge!(options)
+        tag(:link, tag_options(link_options))
+      }.join("\n")
+    end
+
+    def javascript_include_tag(*sources)
+      options = stringify_keys(extract_options!(sources))
+      sources.uniq.map { |source|
+        script_options = {
+            "src" => asset_path(source)
+        }.merge!(options)
+        content_tag(:script, tag_options(script_options))
+      }.join("\n")
+    end
+
+    def content_tag(name, options = nil)
+      "<#{name}#{options}></#{name}>"
+    end
+
+    def tag(name, options)
+      "<#{name}#{options}/>"
+    end
+
+    def extract_options!(options)
+      if options.last.is_a?(Hash) && extractable_options?(options.last)
+        options.pop.except(:integrity, "integrity")
+      else
+        {}
+      end
+    end
+
+    def extractable_options?(options)
+      options.instance_of?(Hash)
+    end
+
+    def stringify_keys(options)
```

Comment:
Because we rely on 'active_support/core_ext/hash' we can call `.stringify_keys` on the hash directly, so we don't need to implement this here.  It's in rails 3.1 so doesn't break our dependency (see: http://guides.rubyonrails.org/v3.1/active_support_core_extensions.html#stringify_keys-and-stringify_keys).